File: ride.cpp
/*
ID: bigfish2
LANG: C++
TASK: ride
*/ 

#include <iostream>
#include <string>
#include <fstream>

#define endl "\n"

using namespace std;

int main(){
    ifstream fin ("ride.in");
    ofstream fout ("ride.out");
    
    string one, two;
    fin >> one >> two;
    int pro = 1, pro1 = 1;
        
    for(int x = 0;x<one.length();x++){
        pro *= one.at(x)-64;
    }    
        
    for(int x = 0;x<two.length();x++){
        pro1 *= two.at(x)-64;
    }    
    if(pro1%47==pro%47) fout << "GO" << endl;
    else fout << "STAY" << endl; 

    return 0;
}

File: ride.java
package Solved;
/*
ID: bigfish2
LANG: JAVA
TASK: ride
*/
import java.io.*;

class ride {
	
  public static void main (String [] args) throws IOException {
    // Use BufferedReader rather than RandomAccessFile; it's much faster
    BufferedReader f = new BufferedReader(new FileReader("ride.in"));
    PrintWriter out = new PrintWriter(new BufferedWriter(new FileWriter("ride.out")));
    
    String comet = f.readLine();
    String group = f.readLine();
    int number = 0;
    int total = 1;
    int modulus;
    
	for(int a = 0; a<comet.length(); a++){
		char holder = comet.charAt(a);
		number = (int) holder;
		number = number - 64;
		total = total * number;
	}
    modulus = total%47;
    
    number = 0;
    total = 1;
    int modulus2;
    
	for(int a = 0; a<group.length(); a++){
		char holder = group.charAt(a);
		number = (int) holder;
		number = number - 64;
		total = total * number;
	}
    modulus2 = total%47;
    
    if (modulus2 == modulus) out.println("GO");
    if (modulus2 != modulus) out.println("STAY");
    
    out.close();
    System.exit(0);                               // don't omit this!
  }
}